src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -187,16 +187,8 @@ class Config:
 
         await self.middleware.run_in_io_thread(self.validate_paths, data, schema_name, verrors)
 
-        if not data["networks"]:
-            verrors.add(f"{schema_name}.networks", "At least one network is required")
-
-        for i, network1 in enumerate(data["networks"]):
-            network1 = ipaddress.IPv4Network(network1, strict=False)
-            for j, network2 in enumerate(data["networks"]):
-                if j > i:
-                    network2 = ipaddress.IPv4Network(network2, strict=False)
-                    if network1.overlaps(network2):
-                        verrors.add(f"{schema_name}.network.{j}", "Networks {network1} and {network2} overlap")
+        if not data["networks"] and not data["hosts"]:
+            verrors.add(f"{schema_name}.networks", "At least one host or network is required")
```

Comment:
@themylogin  It is possible to create a share without hosts or network, any reason we are not allowing it?

Reply:
@william-gr would it be accessible to everyone or no one?

Reply:
@themylogin everyone, as far as I was told.

Reply:
@themylogin @william-gr without any network or host entry, it defaults to everyone